### PR TITLE
Add Join Firefox modal to /firefox/new/ (Fixes #7232)

### LINKED ISSUE #7232: pre-download FxA CTA for Firefox users modal - Code
The copy and design here is final, from https://github.com/mozilla/bedrock/issues/7201

"here's the sketch file with the modal design and final copy FxA-modal-:[new.sketch.zip](https://github.com/mozilla/bedrock/files/3278364/FxA-modal-.new.sketch.zip)"

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Adds a join Firefox modal to /firefox/new/.
  - Modal is shown when clicking "Download Now"
  - Modal is shown only to Firefox users who aren't already signed-in to an account.

## Issue / Bugzilla link
#7232

## Testing
Demo URL: https://www-demo5.allizom.org/en-US/firefox/new/

Successful integration test run: https://ci.vpn1.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/344/pipeline

### CODE REVIEW

**Comment by alexgibson** [comment]
This has now been approved by stakeholders so is good to review/merge.

File: media/css/firefox/new/trailhead/download.scss [hunk 1]
```diff
@@ -292,16 +292,48 @@ html.ios .mzp-c-hero {
...
 
+//* ====================================================================== */
+//  Join Firefox modal
+
+.join-firefox-content {
+    background: $color-white;
+    color: $color-off-black;
+    margin-top: $spacing-lg;
+    text-align: center;
+    padding: $spacing-lg;
```
Comment (stephaniehobson) [inline media/css/firefox/new/trailhead/download.scss]: Nit: alphabetize.

File: media/css/firefox/new/trailhead/download.scss [hunk 2]
```diff
@@ -292,16 +292,48 @@ html.ios .mzp-c-hero {
...
+    }
+}
+
 /**
  * Custom narrow modal style.
  * These should be standardized into a narrow theme.
  * https://github.com/mozilla/protocol/issues/358
  */
-.mzp-c-modal.other-platforms-modal {
+ .mzp-c-modal {
```
Comment (stephaniehobson) [inline media/css/firefox/new/trailhead/download.scss]: Nit: extra space

**Comment by stephaniehobson** [comment]
Restarting before I finish my review.

**Comment by stephaniehobson** [comment]
Didn't find anything else with a signed in user. Code is looking good, just one UX edge case to consider.

**Comment by alexgibson** [comment]
> If I open the other platforms modal and then click download firefox I get the signup modal, and stretched the full width of the page. I think it would be best if the user did not get the Join modal from the download button in the other platforms modal.

Thanks for catching! This is definitely a bug - left over copy / pasta from the old code. Will fix 👍

**Comment by alexgibson** [comment]
Thanks for the review @stephaniehobson updated

**Pull request merged**